src/Omega_h_mesh.hpp

```diff
@@ -125,10 +125,92 @@ class Mesh {
   Graph ask_star(Int dim);
   Graph ask_dual();
 
+/* ask_revClass (Int edim, LOs class_ids): takes input of entity dimension
+ * 'edim', and an 1d array of model entity IDs to return
+ * a CSR structure (Adj) containing IDs of mesh entities classified on the 
+ * requested input model entities. Note here that 'edim' is equal to the
+ * model entity dimension as well as the dimension of returned mesh entities
+ * NOTE: if the model entity is a region, the memory usage is high
+ */
+  Adj ask_revClass (Int edim, LOs class_ids);
+
+/* ask_revClass (Int edim): see ask_revClass (Int edim, LOs class_ids) above.
+ * Here, the output is for all model entities of dimension 'edim' instead
+ * of a input list
+ */
   Adj ask_revClass (Int edim);
+
+/* ask_revClass_downAdj (Int from, Int to): takes input of a higher
+ * dimension 'from' and a lower dimension 'to'. The value of 'from' is equal
+ * to the mesh and model entity dimensions used to get reverse class.
+ * (similar to 'edim' for ask_revClass functions above.) This function can be
+ * understood as a two step process. Firstly, for all the model entities
+ * of dimension 'from', we get the reverse classified mesh entities.
+ * Then combine the reverse classification, and downward adjacency information.
+ * The final output is a CSR containing downward adjacent
+ * mesh entities of dimension 'to' which bound the reverse classified mesh
+ * entities of dimension 'from'.
+ */
+  Adj ask_revClass_downAdj (Int from, Int to);
+
+/* has_revClass (Int edim): Input is a entity dimension 'edim'. This function
+ * checks if the reverse classification for that dimension is present in
+ * memory or not.
+ */
   bool has_revClass (Int edim) const;
+
+/* Takes input of model entity IDs, entity dimension, name of field and number
+ * of components, to create a the rcField. This function
+ * is used when fields are to be stored with mesh entities returned by
+ * ask_revClass (Int edim, LOs class_ids)
+ */
+  template <typename T>
+  void add_rcField(LOs class_ids, Int ent_dim, std::string const& name,
+                   Int ncomps);
+
+/* Takes input of entity dimension, name of rcField, values of rcField, and
+ * stores the values in memory.
+ */
+  template <typename T>
+  void set_rcField_array(Int ent_dim, std::string const& name,
+                         Read<T> array, bool internal = false);
```

Comment:
What does the `internal` argument do?

Reply:
Please look here: https://github.com/SCOREC/omega_h/blob/8f83c0f05671215731bd97baf467571228daf668/src/Omega_h_mesh.cpp#L233-L236

Reply:
Thank you.  It seems like if `internal` is `true` then it will assume that there are no derived/dependent tags that need to be destroyed.  We should add a brief comment about this in the function documentation.  I wonder what the use case is for setting it to `true`.  The comment about 'migration/adaptation/file reading' seems to be a hint... but I don't understand the connection.

Reply:
Yeah, i cant think of any good use case for 'internal=true' right now